### ISSUE TITLE
user selectable range

### ISSUE THREAD

**Issue by blende64** [thegouger/MIDI2LR#25]
Hi, thank you for this nice tool. I have one improvement suggestion:

You are retrieving the range with "local min, max = LrDevelopController.getRange(param)". But it would be a big improvement if the min and max values could be overwritten by user, may be in config file. Especially for Temperature, Tint and Exposure this would be very helpful. If the little steps available are mapped to the full range, each step is much to big. For example Temperature range is 2000 to 50000 so each step is with 400-500k much too rough. But if the range could be mapped to a user defined range, for example 3000-9000 (i have never used anything outside) each step would be about 47 and fine enough.

**Comment by micksh6** [comment]
I agree with this request. I got NanoKontrol2 and I understand that it has 128 gradations for each fader and knob according to MIDI2LR application.
Extending these gradations to full White Balance setting range results in too rough granularity. With mouse I can adjust Temperature with ~100 unit step, but with MIDI2LR - only with ~400 unit step.

At least for white balance it (probably) would be nice to limit the range to current Lightroom setting plus/minus the range MIDI device allows with finest step that Lightroom allows.

For Exposure MIDI2LR already works better than mouse for me. With mouse my smallest step to adjust exposure is 0.12 EV. With MIDI2LR - 0.08-0.07 EV. Excellent! And I got gamer's high resolution mouse. But, having adjustable range wouldn't hurt, of course.

I understand it could be difficult though, given that all this has to be intuitive. I like current implementation of "picking up" settings when moving to the next photo with different LR slider positions. Again, using non-motorized NanoKontrol2 faders.

After my limited testing, everything else works well, as designed. I will post more later.
Kudos to author!

**Comment by pjparra** [comment]
What would be even cooler is being able to assign one control (typically a rotary knob) to change the range, and use another control (typically a fader) to change the value of the setting within that range.

For example, the knob at the zero position would set the range to 2000-6000, at 64 it would set the range to 6000-10000, and at 127 it would set it to 10000-50000.

It would allow us to keep the whole range and having a finer control over the temperature.
It would certainly make the configuration a little trickier though.

**Comment by blende64** [comment]
a nice solution could be adding additional columns in gui like this:
![midi2lr_gui_minmax](https://cloud.githubusercontent.com/assets/1218441/9429843/5fa60798-49de-11e5-87f1-3ec765b96458.gif)
These could be predefined with values from LR and are easy accessible for user.

Currently I just added some constants to the LR plugin. This is already extremely helpful. And if I need a value outside limits I still can set them direct in LR.

**Comment by thegouger** [comment]
Yea, I was considering having the user be able to set the min/max akin to how @blende64 showed.

**Comment by pjparra** [comment]
Yes, @blende64 solution seems to be the best compromise. On second thought, my solution is probably a bit excessively complex, and not so useful, as the temperature range we are using rarely exceeds certain values.

**Comment by PhantomLordM** [comment]
Hello, I agree with @blende64 it would be very useful to be able to select range, especially in temp slider. I wrote email to the @thegouger about my idea of making buttons with selectable step, let's say +50 and -50 so one could fine tune temp setting with a few presses of button. Nevertheless idea presented by @blende64 seems to be easier and better.
The one thing is it should probably work on percentage, like selecting range for example from -45% to +65% of slider (instead of default -100% to +100%) because it should work not only on proprietary RAW files, which allow to set temp in K, but also with TIFF files which allow setting temp from -100 to +100 points and tint the same, instead of 2000 to 50000 (temp) and -150 to +150 (tint) in RAW files.

**Comment by rsjaffe** [comment]
If you want to go even further in the customizability, you could add another column in the app for max value, with a default of 127. While the controls default to 127, on the BCF2000 you could extend it much further. That could help with fine adjustments by making the increments smaller.

Of course, this would have implications for the lua code as well, where it is assumed that the max value will be 127.

**Comment by rsjaffe** [comment]
This issue was moved to rsjaffe/MIDI2LR#10
